src/engine/sparqlExpressions/SparqlExpressionValueGetters.cpp

```diff
@@ -136,3 +136,23 @@ bool IsValidValueGetter::operator()(
   // implement it in a further version.
   return id != ValueId::makeUndefined();
 }
+
+// ____________________________________________________________________________
+IntDoubleStr ToNumericValueGetter::operator()(
+    ValueId id, [[maybe_unused]] const EvaluationContext* context) const {
+  switch (id.getDatatype()) {
+    case Datatype::Int:
+      return id.getInt();
+    case Datatype::Double:
+      return id.getDouble();
+    default:
```

Comment:
You still have to handle the other cases by converting them to strings.
Best you can use the `LiteralFromIdGetter` that returns std::optional<std::string>. If the returned string is nullopt you can return `monostate` and else the string. This is important for the cases like `VocabIndex`, `LocalVocabIndex`, etc.

The different between the two ValueGetters that return strings is that one of them only accepts string literals, while the other one explicitly converts to strings, take

Reply:
Is implemented, also added test cases in which `bool` values and `literals` are given as values.